packages/htmlbars-runtime/lib/expression-visitor.js

```diff
@@ -36,10 +36,12 @@ function acceptExpression(node, env, scope) {
 
   // Primitive literals are unambiguously non-array representations of
   // themselves.
-  if (typeof node !== 'object' || node === null) {
-    ret.value = node;
-  } else {
+  if (Array.isArray(node)) {
+    // if (node.length !== 7) { throw new Error('FIXME: Invalid statement length!'); }
```

Comment:
This was used to ensure that both HTMLBars and Ember's test suites pass without this error being thrown.  Ideally, we would include this as an assertion in the build output, but there is no machinery setup to allow that to be stripped at the moment.